// _@xsl_utils@1.2.6@@xsl/utils/src/medclips.js
import moment from 'moment'
import uuid from 'uuid/v4'
import request from './request'
import { medclips } from './config'
import { isArray } from './lang'

// placeholder暂时不传
// 可以通过指定uid来保证不会追加
export const addLinkContent = async ({ patientId, doctorId, title, placeholder = 'xxx', url, uid }) => {
  if (!patientId || !doctorId) throw new Error('缺少医生id或者患者id')
  if (!url) throw new Error('缺少链接url')
  const finalUid = uid || uuid()
  const requestUrl = `${medclips.contentUrl}/_insecure/v1/${doctorId}/patient/${patientId}/${finalUid}`
  const resp = await request.post(requestUrl, {
    uid: finalUid,
    templateId: 'CUSTOM_MEDICAL_RECORD',
    templateType: 'CUSTOM_MEDICAL_RECORD',
    contentValue: {
      title,
      placeholderImage: placeholder,
      flowIndexUrl: url,
      flowId: '0',
    },
    createdDatetime: moment().format('YYYY-MM-DD HH:mm:ss'),
  }, {
    token: 'random-aea30430-d39b-4a80-84d3-bf2f4f489c76',
  })
  return resp
}

// 根据患者id获取患者填写的表单信息接口

export const getContentInfoByPatientIds = ({ patientIds }) => {
  if (!patientIds || !isArray(patientIds)) throw new Error('患者id提供有误')
  const requestUrl = `${medclips.contentUrl}/patient/item/form/${patientIds.join(',')}`
  return request.get(requestUrl, {}, {
    token: 'random-aea30430-d39b-4a80-84d3-bf2f4f489c76',
  })
}
