// _@xsl_utils@1.2.6@@xsl/utils/lib/config.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var config = {
  development: {
    uas: 'http://qa-uas-management.xsl.link/management/default',
    uasDomain: 'https://qa-uas.xingshulin.com',
    amqpDomains: ['amqp://guest:guest@192.168.64.68:5672', 'amqp://guest:guest@192.168.64.65:5672'],
    smsApiUrl: 'http://qa-sms.xsl.link/sms',
    logger: {
      consoleLevel: 'debug',
      fileLevel: 'error'
    },
    mongodb: 'mongodb://platform2:12345678@192.168.64.65:27017/platform',
    qiniu: {
      ACCESS_KEY: 'a0h5P_uSX_opuu0veh0W3gBYxunaVmIjpgQur9BM',
      SECRET_KEY: 'FdTV3SjhuRGGyLjdHpLosDYhFLimnrcyGV6MpTbG',
      domain: 'https://nzpc6oqy2.qnssl.com',
      feedback: {
        bucket: 'feedback',
        domain: 'https://qa-feedback.xingshulinimg.com',
        zone: 'Zone_z0'
      },
      social: {
        bucket: 'social',
        domain: 'https://qa-social.xingshulinimg.com',
        zone: 'Zone_z0'
      }
    },
    followup: {
      url: 'https://qa-medchart-patient-api.xingshulin.com',
      casefolder: 'http://qa-casefolder-followup-inner.xsl.link',
      message: 'http://qa-followup.xingshulin.com/new-cloudfollowup-service/inner/sendMsgFromWechat'
    },
    medclips: {
      contentUrl: 'http://qa-medchart-content-api.xingshulin.com'
    },
    reward: {
      token: '95a52fb5bf2c5f08803568f2f71f219d',
      secret: '040d94c42fff5af7bd6e82fc5f93afc4'
    },
    bmDb: {
      client: 'mysql2',
      connection: {
        host: '192.168.64.66',
        port: 33306,
        user: 'buddy-manager',
        password: 'buddy-manager',
        database: 'buddy_manager'
      },
      pool: {
        min: 10,
        max: 30
      },
      acquireConnectionTimeout: 3000
    },
    drDomain: 'http://qa-dr.xingshulin.com',
    feedBack: {
      adminId: '2024733'
    },
    notice: {
      Authorization:
      // eslint-disable-next-line
      'Bearer eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJzZXJ2aWNlOm1hbmFnZW1lbnQifQ.I_bcNUD_OKBQfSrx5MoH5hTFPVVFqRQVE9YG3GX-UAAXR3zo9xQCb8XUZkf2U5KUMN67TkvJucdCU3ISXIXycQ',
      Domain: 'http://qa-xmb.xingshulin.com'
    }
  },
  test: {
    uas: 'http://qa-uas-management.xsl.link/management/default',
    uasDomain: 'https://qa-uas.xingshulin.com',
    amqpDomains: ['amqp://guest:guest@192.168.64.68:5672', 'amqp://guest:guest@192.168.64.65:5672'],
    smsApiUrl: 'http://qa-sms.xsl.link/sms',
    logger: {
      consoleLevel: 'debug',
      fileLevel: 'error'
    },
    mongodb: 'mongodb://platform2:12345678@192.168.64.65:27017/platform',
    qiniu: {
      ACCESS_KEY: 'a0h5P_uSX_opuu0veh0W3gBYxunaVmIjpgQur9BM',
      SECRET_KEY: 'FdTV3SjhuRGGyLjdHpLosDYhFLimnrcyGV6MpTbG',
      domain: 'https://nzpc6oqy2.qnssl.com',
      feedback: {
        bucket: 'feedback',
        domain: 'https://qa-feedback.xingshulinimg.com',
        zone: 'Zone_z0'
      },
      social: {
        bucket: 'social',
        domain: 'https://qa-social.xingshulinimg.com',
        zone: 'Zone_z0'
      }
    },
    followup: {
      url: 'https://qa-medchart-patient-api.xingshulin.com',
      casefolder: 'http://qa-casefolder-followup-inner.xsl.link',
      message: 'http://qa-followup.xingshulin.com/new-cloudfollowup-service/inner/sendMsgFromWechat'
    },
    medclips: {
      contentUrl: 'http://qa-medchart-content-api.xingshulin.com'
    },
    reward: {
      token: '95a52fb5bf2c5f08803568f2f71f219d',
      secret: '040d94c42fff5af7bd6e82fc5f93afc4'
    },
    bmDb: {
      client: 'mysql2',
      connection: {
        host: '192.168.64.66',
        port: 33306,
        user: 'buddy-manager',
        password: 'buddy-manager',
        database: 'buddy_manager'
      },
      pool: {
        min: 10,
        max: 30
      },
      acquireConnectionTimeout: 3000
    },
    drDomain: 'http://qa-dr.xingshulin.com',
    feedBack: {
      adminId: '2024733'
    },
    notice: {
      Authorization:
      // eslint-disable-next-line
      'Bearer eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJzZXJ2aWNlOm1hbmFnZW1lbnQifQ.I_bcNUD_OKBQfSrx5MoH5hTFPVVFqRQVE9YG3GX-UAAXR3zo9xQCb8XUZkf2U5KUMN67TkvJucdCU3ISXIXycQ',
      Domain: 'http://qa-xmb.xingshulin.com'
    }
  },
  production: {
    uas: 'http://uas-management.xsl.link/management/default',
    uasDomain: 'http://uas.xsl.link',
    amqpDomains: ['amqp://guest:guest@192.168.20.100:5672', 'amqp://guest:guest@192.168.20.101:5672', 'amqp://guest:guest@192.168.20.102:5672'],
    smsApiUrl: 'http://sms.xsl.link/sms',
    logger: {
      consoleLevel: 'info',
      fileLevel: 'error',
      filename: '/logs/wxpage/wxpage.log'
    },
    mongodb: 'mongodb://platform:43&lGsa@192.168.20.163:27017/platform',
    qiniu: {
      ACCESS_KEY: '1AXRtAuXagE1dTL3onHnV9vP262l7ZD-J8VjZd0W',
      SECRET_KEY: '31jUIkyvI8w-JMEPLuG5Ix5tk4mXzq3vw_sNCJan',
      domain: 'https://social.xingshulinimg.com',
      feedback: {
        bucket: 'feedback',
        domain: 'https://feedback.xingshulinimg.com',
        zone: 'Zone_z0'
      },
      social: {
        bucket: 'social',
        domain: 'https://social.xingshulinimg.com',
        zone: 'Zone_z0'
      }
    },
    followup: {
      url: 'http://medchart-patient-api.xsl.link',
      casefolder: 'http://casefolder-followup-inner.xsl.link',
      message: 'http://followup.xsl.link/new-cloudfollowup-service/inner/sendMsgFromWechat'
    },
    medclips: {
      contentUrl: 'http://medchart-content-api.xsl.link'
    },
    reward: {
      token: 'c805780a5c9b5bad92f86eda4ca73518',
      secret: 'f1f9e02784f0546788c63fd94384b259'
    },
    bmDb: {
      client: 'mysql2',
      connection: {
        host: '192.168.20.124',
        port: 33306,
        user: 'buddy-manager',
        password: 'xmF9yMpvrC',
        database: 'buddy_manager'
      },
      pool: {
        min: 10,
        max: 30
      },
      acquireConnectionTimeout: 3000
    },
    drDomain: 'http://dr.xingshulin.com',
    feedBack: {
      adminId: '2290078'
    },
    notice: {
      Authorization:
      // eslint-disable-next-line
      'Bearer eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJzZXJ2aWNlOm1hbmFnZW1lbnQifQ.vbWvpkW6CIjuKCxE6bPZgH9iLsRx0x9ngWKIGxpYh8CReTwtpdPbnJj4iZXvIT9CgyrxsZ27rd8L0AKxG8SOjA',
      Domain: 'https://xmb.xingshulin.com'
    }
  }
};

var env = 'development';
if (process.env.EGG_SERVER_ENV === 'prod' || process.env.EGG_SERVER_ENV === 'production') {
  env = 'production';
} else {
  env = process.env.EGG_SERVER_ENV || process.env.NODE_ENV || 'development';
}

exports.default = config[env];
module.exports = exports['default'];
