// _@xsl_utils@1.2.6@@xsl/utils/lib/dbutils.js
'use strict';

var _toConsumableArray2 = require('babel-runtime/helpers/toConsumableArray');

var _toConsumableArray3 = _interopRequireDefault(_toConsumableArray2);

var _stringify = require('babel-runtime/core-js/json/stringify');

var _stringify2 = _interopRequireDefault(_stringify);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var R = require('ramda');
var lang = require('./lang');

function stringify(o) {
  if (lang.isObject(o) || lang.isArray(o)) {
    try {
      return (0, _stringify2.default)(o);
    } catch (e) {
      // not handle
    }
  }
  return o;
}

var trans = function trans(transKey, transValue, obj) {
  return R.reduce(function (acc, key) {
    acc[transKey(key)] = transValue(obj[key]);
    return acc;
  }, {}, R.keys(obj));
};

var toLower = R.compose(R.replace(/([A-Z])/g, function (_, s) {
  return '_' + R.toLower(s);
}), R.replace(/^([A-Z])/, function (_, s) {
  return R.toLower(s);
}) // 第一个字母直接 toLower
);
function transformObjToDb(obj) {
  return trans(toLower, stringify, obj);
}

var toUpper = R.replace(/_(\w)/g, function (_, s) {
  return R.toUpper(s);
});
function transformDbToObj(obj) {
  return trans(toUpper, function (e) {
    return e;
  }, obj);
}

function zipAll(fun) {
  for (var _len = arguments.length, arrs = Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
    arrs[_key - 1] = arguments[_key];
  }

  var len = Math.min.apply(Math, (0, _toConsumableArray3.default)(arrs.map(function () {
    var arr = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : [];
    return arr.length || 0;
  })));
  return R.repeat(1, len).map(function (_, index) {
    return fun.apply(undefined, (0, _toConsumableArray3.default)(arrs.map(function () {
      var arr = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : [];
      return arr[index];
    })));
  });
}

module.exports = {
  transformObjToDb: transformObjToDb,
  transformDbToObj: transformDbToObj,
  zipAll: zipAll,
  toUpper: toUpper,
  toLower: toLower
};
