// _@xsl_utils@1.2.6@@xsl/utils/lib/lang.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.mobileRegex = exports.emailRegex = undefined;

var _keys = require('babel-runtime/core-js/object/keys');

var _keys2 = _interopRequireDefault(_keys);

exports.isArray = isArray;
exports.isFunction = isFunction;
exports.isObject = isObject;
exports.isString = isString;
exports.isEmpty = isEmpty;
exports.isUrl = isUrl;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * 此工具方法中主要存放一些 js 对象或者语言方面的处理函数
 * PS: lang 是 language 的简称
 */

function isArray(o) {
  return Object.prototype.toString.apply(o) === '[object Array]';
}

function isFunction(o) {
  return Object.prototype.toString.apply(o) === '[object Function]';
}

function isObject(o) {
  return Object.prototype.toString.apply(o) === '[object Object]';
}

function isString(o) {
  return Object.prototype.toString.apply(o) === '[object String]';
}

function isEmpty(value) {
  return value === undefined || isString(value) && value.trim() === '' || isArray(value) && value.length === 0 || isObject(value) && (0, _keys2.default)(value).length === 0;
}

function isUrl(value) {
  return (/^https?:\/\/.*/.test(value)
  );
}

var emailRegex = exports.emailRegex = /^[a-z0-9A-Z]+([._\\\\-]*[a-z0-9A-Z])*@([a-z0-9A-Z]+[-a-z0-9A-Z]*[a-z0-9A-Z]+.){1,63}[a-z0-9A-Z]+$/;

var mobileRegex = exports.mobileRegex = /^1[3|4|5|7|8][0-9]\d{8}$/;
