// _@xsl_utils@1.2.6@@xsl/utils/lib/platform.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.isIOS = isIOS;
exports.isIOS9 = isIOS9;
exports.isIOS8 = isIOS8;
exports.isMobile = isMobile;
exports.isAndroid = isAndroid;
exports.androidVersion = androidVersion;
exports.hasWebSocket = hasWebSocket;
exports.isWeiXin = isWeiXin;
exports.isQQ = isQQ;
exports.isIE = isIE;
exports.closeKeyboard = closeKeyboard;
exports.isIphoneX = isIphoneX;
exports.isIphone5 = isIphone5;
/**
 * 仅前端适用
 * 此工具方法中主要存放前端页面判断环境的一些方法
 */
/* eslint-disable */
function isIOS() {
  var ua = window.navigator.userAgent;
  return !!ua.match(/\(i[^;]+;( U;)? CPU.+Mac OS X/);
}

function isIOS9() {
  return navigator.userAgent.indexOf('iPhone OS 9') > -1;
}

function isIOS8() {
  return navigator.userAgent.indexOf('iPhone OS 8') > -1;
}

function isMobile() {
  var ua = window.navigator.userAgent.toLowerCase();

  return !!ua.match(/applewebkit.*mobile.*/);
}

function isAndroid() {
  var ua = window.navigator.userAgent.toLowerCase();
  // android终端或者uc浏览器
  return ua.indexOf('android') > -1 || ua.indexOf('linux') > -1;
}

function androidVersion() {
  var ua = window.navigator.userAgent.toLowerCase();
  return parseFloat(ua.substr(ua.search('android') + 8, 5));
}

function hasWebSocket() {
  return window && !(!window.WebSocket || androidVersion() < 4.4);
}

function isWeiXin() {
  var ua = navigator.userAgent.toLowerCase();
  return !!/micromessenger/.test(ua);
}

function isQQ() {
  var reg = new RegExp('qq/', 'g');
  var ua = navigator.userAgent.toLowerCase();
  return !!reg.test(ua);
}

function isIE() {
  return !!window.ActiveXObject || 'ActiveXObject' in window;
}

function closeKeyboard() {
  document.activeElement.blur();
  var inputs = document.querySelectorAll('input');
  for (var i = 0; i < inputs.length; i++) {
    inputs[i].blur();
  }
}

function isIphoneX() {
  return isIOS() && window.screen.height === 812;
}

function isIphone5() {
  return isIOS() && window.screen.width === 320;
}

/* eslint-enable */
