// _@xsl_utils@1.2.6@@xsl/utils/lib/send-notice.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _regenerator = require('babel-runtime/regenerator');

var _regenerator2 = _interopRequireDefault(_regenerator);

var _asyncToGenerator2 = require('babel-runtime/helpers/asyncToGenerator');

var _asyncToGenerator3 = _interopRequireDefault(_asyncToGenerator2);

// http://wiki.xingshulin.com/pages/viewpage.action?pageId=51713361#tag/%E4%B8%9A%E5%8A%A1%E6%8E%A5%E5%85%A5
// messageTagId分别为1：病历审核，2：随访申请，3：项目通知，4：钱包助手，5：只能助手，6：系统通知，7：好友申请，8：公告通知
var sendNormalNotice = function () {
  var _ref = (0, _asyncToGenerator3.default)( /*#__PURE__*/_regenerator2.default.mark(function _callee(_ref2) {
    var appNames = _ref2.appNames,
        userIds = _ref2.userIds,
        _ref2$messageTagId = _ref2.messageTagId,
        messageTagId = _ref2$messageTagId === undefined ? 3 : _ref2$messageTagId,
        _ref2$priority = _ref2.priority,
        priority = _ref2$priority === undefined ? 10 : _ref2$priority,
        content = _ref2.content;
    return _regenerator2.default.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            _context.next = 2;
            return _request2.default.post(Domain + '/message/v2/delivery', {
              dest: {
                appName: appNames,
                userId: userIds
              },
              priority: priority,
              messageTagId: messageTagId,
              content: content
            }, {
              Authorization: Authorization,
              'X-Request-From': 'service:management',
              'Content-Type': 'application/json'
            });

          case 2:
          case 'end':
            return _context.stop();
        }
      }
    }, _callee, this);
  }));

  return function sendNormalNotice(_x) {
    return _ref.apply(this, arguments);
  };
}();

var _request = require('./request');

var _request2 = _interopRequireDefault(_request);

var _config = require('./config');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Authorization = _config.notice.Authorization,
    Domain = _config.notice.Domain;
exports.default = sendNormalNotice;
module.exports = exports['default'];
