// _@xsl_utils@1.2.6@@xsl/utils/lib/uas.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.temp = exports.getUserInfoByMobile = exports.getUserInfoByIds = undefined;

var _regenerator = require('babel-runtime/regenerator');

var _regenerator2 = _interopRequireDefault(_regenerator);

var _defineProperty2 = require('babel-runtime/helpers/defineProperty');

var _defineProperty3 = _interopRequireDefault(_defineProperty2);

var _extends3 = require('babel-runtime/helpers/extends');

var _extends4 = _interopRequireDefault(_extends3);

var _promise = require('babel-runtime/core-js/promise');

var _promise2 = _interopRequireDefault(_promise);

var _asyncToGenerator2 = require('babel-runtime/helpers/asyncToGenerator');

var _asyncToGenerator3 = _interopRequireDefault(_asyncToGenerator2);

var _ramda = require('ramda');

var _ramda2 = _interopRequireDefault(_ramda);

var _request = require('./request');

var _request2 = _interopRequireDefault(_request);

var _config = require('./config');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var getUserInfoByIds = exports.getUserInfoByIds = function () {
  var _ref = (0, _asyncToGenerator3.default)( /*#__PURE__*/_regenerator2.default.mark(function _callee(userIds) {
    var fields = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : [];
    var url, userIdArrs, resp, userInfoMap;
    return _regenerator2.default.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            url = _config.uas + '/search/user/';
            userIdArrs = _ramda2.default.splitEvery(20, userIds);
            _context.next = 4;
            return _promise2.default.all(userIdArrs.map(function (ids) {
              return _request2.default.post(url, {
                userIds: ids,
                // fields: ['schools', 'hospitals', 'areaId', 'introduce', 'lastRealnameTimestamp'],
                fields: fields
              });
            }));

          case 4:
            resp = _context.sent;
            userInfoMap = _ramda2.default.flatten(resp.map(_ramda2.default.prop('data'))).reduce(function (f, v) {
              return (0, _extends4.default)({}, f, (0, _defineProperty3.default)({}, v.userId, v));
            }, {});
            return _context.abrupt('return', userInfoMap);

          case 7:
          case 'end':
            return _context.stop();
        }
      }
    }, _callee, undefined);
  }));

  return function getUserInfoByIds(_x) {
    return _ref.apply(this, arguments);
  };
}();

var getUserInfoByMobile = exports.getUserInfoByMobile = function () {
  var _ref2 = (0, _asyncToGenerator3.default)( /*#__PURE__*/_regenerator2.default.mark(function _callee2(mobile) {
    var fields = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : ['schools', 'hospitals', 'areaId', 'introduce', 'lastRealnameTimestamp'];
    var url, resp, userInfo;
    return _regenerator2.default.wrap(function _callee2$(_context2) {
      while (1) {
        switch (_context2.prev = _context2.next) {
          case 0:
            url = _config.uas + '/search/user/list';
            _context2.next = 3;
            return _request2.default.post(url, {
              critical: [{
                field: 'account',
                type: 'term',
                value: mobile
              }],
              fields: fields
            });

          case 3:
            resp = _context2.sent;
            userInfo = _ramda2.default.path(['data', 0], resp);
            return _context2.abrupt('return', userInfo);

          case 6:
          case 'end':
            return _context2.stop();
        }
      }
    }, _callee2, undefined);
  }));

  return function getUserInfoByMobile(_x3) {
    return _ref2.apply(this, arguments);
  };
}();

var temp = exports.temp = null;
