// _@xsl_utils@1.2.6@@xsl/utils/lib/wechat/sessionHeader.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _regenerator = require('babel-runtime/regenerator');

var _regenerator2 = _interopRequireDefault(_regenerator);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _asyncToGenerator2 = require('babel-runtime/helpers/asyncToGenerator');

var _asyncToGenerator3 = _interopRequireDefault(_asyncToGenerator2);

var _v = require('uuid/v4');

var _v2 = _interopRequireDefault(_v);

var _request = require('../request');

var _request2 = _interopRequireDefault(_request);

var _config = require('../config');

var _config2 = _interopRequireDefault(_config);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = (0, _asyncToGenerator3.default)( /*#__PURE__*/_regenerator2.default.mark(function _callee() {
  var platform, deviceId, ua, header, _ref2, data;

  return _regenerator2.default.wrap(function _callee$(_context) {
    while (1) {
      switch (_context.prev = _context.next) {
        case 0:
          platform = 'universal-form';
          deviceId = (0, _v2.default)();
          ua = 'www/1.0.0 (' + platform + ', unknown unknown) net/null';
          header = {
            'X-Security-Id': deviceId,
            'X-User-Agent': ua
          };
          _context.next = 6;
          return _request2.default.post(_config2.default.uasDomain + '/passport/default/context', {
            app: {
              name: 'www',
              version: '1.0.0',
              hotfixVersion: '0'
            },
            device: {
              deviceId: deviceId,
              name: 'universal-form',
              systemName: platform,
              systemVersion: '',
              model: '',
              localizedModel: ''
            }
          }, header);

        case 6:
          _ref2 = _context.sent;
          data = _ref2.data;
          return _context.abrupt('return', (0, _extends3.default)({}, header, {
            'X-User-Token': data
          }));

        case 9:
        case 'end':
          return _context.stop();
      }
    }
  }, _callee, undefined);
})); // 服务端生成未登录的sessionkey

module.exports = exports['default'];
