// _@xsl_utils@1.2.6@@xsl/utils/lib/wechat/wxConfig.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var config = {
  tokenUrl: 'https://api.weixin.qq.com/cgi-bin/token',
  ticketUrl: 'https://api.weixin.qq.com/cgi-bin/ticket/getticket',
  apiList: ['onMenuShareTimeline', 'onMenuShareAppMessage', 'onMenuShareQQ', 'onMenuShareWeibo', 'onMenuShareQZone', 'chooseImage', 'uploadImage'],
  logger: console.log

  /**
   *
   * @param appId
   * @param appSecret
   * @param domain
   * @param apiList
   */
};var setWxConfig = exports.setWxConfig = function setWxConfig(_ref) {
  var appId = _ref.appId,
      appSecret = _ref.appSecret,
      domain = _ref.domain,
      apiList = _ref.apiList;

  if (!appId || !appSecret || !domain) {
    throw new Error('注意, 必须提供 appId, appSecret, domain 信息!');
  }
  config.appId = appId;
  config.appSecret = appSecret;
  config.domain = domain;
  config.apiList = apiList || config.apiList;
};

var setWxLogger = exports.setWxLogger = function setWxLogger(logger) {
  config.logger = logger || config.logger;
};

exports.default = config;
